app/vmctl/opentsdb.go

```diff
@@ -0,0 +1,168 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"sync"
+	"time"
+
+	"github.com/VictoriaMetrics/VictoriaMetrics/app/vmctl/opentsdb"
+	"github.com/VictoriaMetrics/VictoriaMetrics/app/vmctl/vm"
+	"github.com/cheggaaa/pb/v3"
+)
+
+type otsdbProcessor struct {
+	oc      *opentsdb.Client
+	im      *vm.Importer
+	otsdbcc int
+	vmcc    int
```

Comment:
I don't understand how it will be used :-)

Reply:
This is the function that is called in main.go to start the loop. Am I doing this wrong?

Reply:
If you're explicitly talking about otsdbcc, https://github.com/johnseekins/VictoriaMetrics/blob/otsdbmig/app/vmctl/opentsdb.go#L87 <- I spin up threads to process the many individual queries that make up a retention query for a metric/series. Because I've segmented each series into a set of smaller queries (to reduce cluster load and - hopefully - potential for stuck threads) the threading happens to make each of these individual smaller queries.

Reply:
Sorry for not being clear. I'm talking specifically about `vmcc` - looks like it is unused currently.